src/analysis/ExpressionExplorer.jl

```diff
@@ -495,6 +495,8 @@ function explore!(ex::Expr, scopestate::ScopeState)::SymbolsState
         else
             return explore!(Expr(:block, ex.args...), scopestate)
         end
+    elseif Meta.isexpr(ex, :parameters)
+        return mapfoldl(a -> explore!(to_kw(a), scopestate), union!, ex.args, init = SymbolsState())
```

Comment:
@Pangoraw Bit unrelated and 100% out of the scope of this PR, but do you think we should parse the expression earlier and convert it to a parametric type such as `Expr{:parameters}` so that we can use multiple dispatch instead of all the `isexpr` checks inside `ExpressionExplorer`?

Reply:
Related: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/

Reply:
interesting, so the advantages of a parametric type would be to improve the code by splitting the if in multiple functions or something?

Reply:
Yes. Would be a tiny bit faster, more readable and more extensible in case someone wants to add new dispatches. It could also reduce TTFX because it would improve inferability.